### PR TITLE
Add pan-arctic atmosphere RRM grid

### PR DESCRIPTION
**Pull request by brhillman**
Add a new pan-arctic atmosphere RRM grid (arcticx4v1pg2). Configuration and input files provided to run with four different configurations:
```
arcticx4v1pg2_oARRM60to10
arcticx4v1pg2_r05_oARRM60to10
arcticx4v1pg2_r125_oARRM60to10
```
[BFB]

### CODE REVIEW

**Comment by wlin7** [comment]
Hi @brhillman , The PR is looking good overall. Though there are some issues (listed below) need your further attention, an F-case is ok to run.

1. map_arcticx4v1pg2_to_r0125_bilin.20210517.nc has frac_a values all in the order of e+13 (as reported by ncks --check_map); map_arcticx4v1pg2_to_oARRM60to10_bilin.20210517.nc has max(frac_b) > 1.5. Can these be tolerated?
2.  arcticx4v1pg2_r05_oARRM60to10 grid is also introduced in this PR, but LND2ROF_FMAPNAME and ROF2LND_FMAPNAME are not defined for rof_grid=r05. Note that the PR does define these maps for rof_grid=r0125.
3. ROF2OCN_ICE_RMAPNAME and ROF2OCN_LIQ_RMAPNAME are not defined. As a result, they are set to idmap. For F-case, it does not matter and the model can still run. For coupled case, they would matter.
4. V2 MPASSI uses config_nSnowLayers=5, but sea ice initial file for oARRM60to10,  inputdata/ice/mpas-cice/oARRM60to10/seaice.ARRM60to10.restart_grizzly.200422.nc, has nSnowLayer=1. That is an old spun-up sea ice file. Please also see issue [#3894](https://github.com/E3SM-Project/E3SM/issues/3894), Either going with no sea ice, or specify a new spun-up sea ice file. Note that the current init template file is specified in components/mpas-seaice/cime_config/buildnml
5. The ATM_NCPL value and you are probably already aware. Just an FYI and no change is necessary to reset the value. V2 RRM chooses to have ATM_NCPL be the same as that for the lowest resolution in the mesh. Previously (V1), ATM_NCPL would be set to match the finest resolution, e.g., explicitly set to 96 for 100 km - 25 km RRM - same as for ne120. The value for this parameter is set in  driver-mct/cime_config/config_component_e3sm.xml. Since no value is specified for arcticx4v1, the default value (48) is used, which happens to be the value used for ne30.

**Comment by brhillman** [comment]
Thanks @wlin7. Some responses:
  1. Perhaps the best thing to do is to switch to `mono` maps for these. Any reason not to do that?
  2. Added ROF2LND and LND2ROF for r05.
  3. Actually, these maps are already defined:
  ```
      <map name="ROF2OCN_ICE_RMAPNAME">cpl/cpl6/map_rx1_to_oARRM60to10_smoothed.r150e300.180718.nc</map>
      <map name="ROF2OCN_LIQ_RMAPNAME">cpl/cpl6/map_rx1_to_oARRM60to10_smoothed.r150e300.180718.nc</map>
      <map name="ROF2OCN_ICE_RMAPNAME">cpl/cpl6/map_JRA025_to_oARRM60to10_smoothed.r150e300.181219.nc</map>
      <map name="ROF2OCN_LIQ_RMAPNAME">cpl/cpl6/map_JRA025_to_oARRM60to10_smoothed.r150e300.181219.nc</map>
      <map name="ROF2OCN_ICE_RMAPNAME">cpl/cpl6/map_r05_to_ARRM60to10_smoothed.r150e300.200413.nc</map>
      <map name="ROF2OCN_LIQ_RMAPNAME">cpl/cpl6/map_r05_to_ARRM60to10_smoothed.r150e300.200413.nc</map>
```
  4. I believe this is planned to be resolved in a subsequent PR.
  5. I actually think the setting for ATM_NCPL is correct. For F-case, ATM_NCPL=48 is sufficient, but for WCYCL cases with oARRM60to10, the ocean and ice models call for a short coupling interval for stability in those components (ATM_NCPL=96). If we set ATM_NCPL=48 explicitly for all arcticx4v1 configs, this conflicts with oARRM60to10 and we get a configuration error:
  ```
  Finished SETUP for test SMS.arcticx4v1pg2_oARRM60to10.WCYCL1850.cori-knl_intel in 6.564114 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /global/cscratch1/sd/bhillma/e3sm_scratch/cori-knl/SMS.arcticx4v1pg2_oARRM60to10.WCYCL1850.cori-knl_intel.20220214_144651_eala9e
    Errors were:
        ERROR: Active atm should match shortest model timestep atmdt=1800 mindt=900

Waiting for tests to finish
FAIL SMS.arcticx4v1pg2_oARRM60to10.WCYCL1850.cori-knl_intel (phase SETUP)
    Case dir: /global/cscratch1/sd/bhillma/e3sm_scratch/cori-knl/SMS.arcticx4v1pg2_oARRM60to10.WCYCL1850.cori-knl_intel.20220214_144651_eala9e
test-scheduler took 10.994227409362793 seconds
```

**Comment by wlin7** [comment]
On #3 above, I missed the ROF2OCN maps from r05 to ARRM60to10. We are still missing ROF2OCN from r0125 to oARRM60to10, if using the other two new grids: arcticx4v1pg2_oARRM60to10 or arcticx4v1pg2_r0125_oARRM60to10. 

Lower priority is only to use arcticx4v1pg2_r05_oARRM60to10 for now.

**Comment by brhillman** [comment]
@wlin7 thanks, I believe @darincomeau is working on producing the ROF2OCN maps for r0125. We can wait and see if that's successful, and if not we can revert the articx4v1pg2_oARRM60to10 config to use r05 for runoff instead of r0125.

**Comment by milenaveneziani** [comment]
Besides the sea ice initial condition file, there is another default-change that needs to be made for this configuration, and it's related to an ocean file (the MOC southern transect file). Would it make sense to include these two changes here, or should they be included in a separate PR?

**Comment by darincomeau** [comment]
@milenaveneziani I'm creating a separate PR for changing the sea ice/ocn initial condition for ARRM60to10, since it also affects other configurations using that mesh.

**Comment by milenaveneziani** [comment]
Sounds great, thanks @darincomeau.

**Comment by darincomeau** [comment]
@brhillman  - @jonbob kindly made the needed `ROF2OCN_ICE_RMAPNAME` / `ROF2OCN_LIQ_RMAPNAME` mapping file (thanks!), and it's on the inputdata server here:
`/lcrc/group/acme/public_html/inputdata/cpl/cpl6/map_r0125_to_ARRM60to10_smoothed.r50e100.220214.nc`

**Comment by brhillman** [comment]
@darincomeau @jonbob great, thanks! I will push a commit to add this.

**Comment by darincomeau** [comment]
@brhillman this PR changes the default sea ice and ocean ICs for the `ARRM60to10` mesh B-cases, in case you need it to test here: https://github.com/E3SM-Project/E3SM/pull/4795

**Comment by rljacob** [comment]
Notes from IG telecon:  needs an additional commit.

File: components/eam/bld/namelist_files/namelist_defaults_eam.xml [hunk 1]
```diff
@@ -101,6 +102,7 @@
 
 <ncdata dyn="se" hgrid="ne0np4_antarcticax4v1" nlev="72"    >atm/cam/inic/homme/cami_mam3_Linoz_0000-01-antarcticax4v1_L72_topoadj_c20200507.nc</ncdata>
 
+<ncdata dyn="se" hgrid="ne0np4_arcticx4v1" nlev="72"    >atm/cam/inic/homme/cami_mam3_Linoz_0000-01-arcticx4v1np4_L72_c20210429.nc</ncdata>
```
Comment (wlin7) [inline components/eam/bld/namelist_files/namelist_defaults_eam.xml]: This IC was a remapping from ne120np4; no sign of topo adjustment. Has it been tested that the model can run stably with it?

File: cime_config/config_grids.xml [hunk 2]
```diff
@@ -3453,6 +3474,21 @@
...
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/arcticx4v1pg2/map_arcticx4v1pg2_to_oARRM60to10_mono.20210407.nc</map>
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/arcticx4v1pg2/map_arcticx4v1pg2_to_oARRM60to10_bilin.20210517.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/arcticx4v1pg2/map_arcticx4v1pg2_to_oARRM60to10_bilin.20210517.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/oARRM60to10/map_oARRM60to10_to_arcticx4v1pg2_mono.20210407.nc</map>
+      <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/oARRM60to10/map_oARRM60to10_to_arcticx4v1pg2_mono.20210407.nc</map>
+    </gridmap>
+
+    <gridmap atm_grid="ne0np4_arcticx4v1.pg2" lnd_grid="r0125">
+      <map name="ATM2LND_FMAPNAME">cpl/gridmaps/arcticx4v1pg2/map_arcticx4v1pg2_to_r0125_mono.20210517.nc</map>
+      <map name="ATM2LND_SMAPNAME">cpl/gridmaps/arcticx4v1pg2/map_arcticx4v1pg2_to_r0125_bilin.20210517.nc</map>
```
Comment (wlin7) [inline cime_config/config_grids.xml]: @brhillman , is it still in the plan to replace in particular this bilin map?

**Comment by wlin7** [comment]
Merged to master.

**Pull request merged**